### DISCUSSION TITLE
How to achieve runtime data loading according to user's input?

### DISCUSSION THREAD

**Discussion by riajain0412** [unicode-org/icu4x#2967] [Q&A]
So, If I want to avoid data bundle and instead want the data file to get loaded as per the user's input. Suppose we have a segmenter code and user inputs German string so I want runtime data loading of german only files. How to achieve this in ICU4X?
I explored icu_provider_adapters but unable to get that how to use this in above stated scenario.

**Comment by Manishearth** [comment]
Make a new data provider for each locale, and then cache it somewhere.

There are a lot of ways to do this: it really depends on the architecture of your application. ICU4X is flexible in letting you choose how you want to do things.

**Reply by sffc** [reply]
You should use a [forking provider](https://icu4x.unicode.org/doc/icu_provider_adapters/fork/index.html) and load in your additional postcard files at runtime when a new locale is requested.

There should be a tutorial for this. I'll convert this into an issue to track that.

**Reply by riajain0412** [reply]
Okay, what exactly is postcard file? Can we also use blob instead of postcard?

**Reply by sffc** [reply]
"postcard file" == "data blob"

**Reply by sffc** [reply]
Issue to track a tutorial for this: https://github.com/unicode-org/icu4x/issues/2970

**Reply by riajain0412** [reply]
Okay, Thank you so much. Let me know once your tutorial is ready.
